src/rdkafka_msgset_reader.c

```diff
@@ -1088,40 +1132,8 @@ rd_kafka_msgset_reader_v2 (rd_kafka_msgset_reader_t *msetr) {
                                               &save_slice, payload_size))
                         rd_kafka_buf_check_len(rkbuf, payload_size);
 
-                if (msetr->msetr_aborted_txns != NULL &&
-                    (msetr->msetr_v2_hdr->Attributes &
-                     (RD_KAFKA_MSGSET_V2_ATTR_TRANSACTIONAL|
-                      RD_KAFKA_MSGSET_V2_ATTR_CONTROL)) ==
-                    RD_KAFKA_MSGSET_V2_ATTR_TRANSACTIONAL) {
-                        /* Transactional non-control MessageSet:
-                         * check if it is part of an aborted transaction. */
-                        int64_t txn_start_offset =
-                                rd_kafka_aborted_txns_get_offset(
-                                        msetr->msetr_aborted_txns,
-                                        msetr->msetr_v2_hdr->PID);
-
-                        if (txn_start_offset != -1 &&
-                            msetr->msetr_v2_hdr->BaseOffset >=
-                            txn_start_offset) {
-                                /* MessageSet is part of aborted transaction */
-                                rd_rkb_dbg(msetr->msetr_rkb, MSG, "MSG",
-                                           "%s [%"PRId32"]: "
-                                           "Skipping %"PRId32" message(s) "
-                                           "in aborted transaction "
-                                           "at offset %"PRId64,
-                                           rktp->rktp_rkt->rkt_topic->str,
-                                           rktp->rktp_partition,
-                                           msetr->msetr_v2_hdr->RecordCount,
-                                           txn_start_offset);
-                                rd_kafka_buf_skip(rkbuf, payload_size);
-                                rd_slice_widen(&rkbuf->rkbuf_reader,
-                                               &save_slice);
-                                goto done;
-                        }
-                }
-
                 /* Read messages */
-                err = rd_kafka_msgset_reader_msgs_v2(msetr);
+                err = rd_kafka_msgset_read_msgs_v2(msetr, payload_size);
```

Comment:
Don't think you need to pass payload_size, the slice is already narrowed to that same size at line 1131

Reply:
Or we could just not rename it, always an option!

Reply:
you mean i should use `rd_slice_remains` to work this out instead?

(note that i was just reorganizing existing logic to arrive at the current code)

Reply:
Yep